lib/msal-core/src/ScopeSet.ts

```diff
@@ -141,15 +130,49 @@ export class ScopeSet {
      * Appends extraScopesToConsent if passed
      * @param {@link AuthenticationParameters}
      */
-    static appendScopes(reqScopes: Array<string>, reqExtraScopesToConsent: Array<string>): Array<string> {
-        if (reqScopes) {
-            const convertedExtraScopes = reqExtraScopesToConsent ? this.trimAndConvertArrayToLowerCase([...reqExtraScopesToConsent]) : null;
+    static appendScopes(reqScopes: Array<string>, scopesToAppend: Array<string>): Array<string> {
+        if(reqScopes) {
+            const convertedScopesToAppend = scopesToAppend ? this.trimAndConvertArrayToLowerCase([...scopesToAppend]) : null;
             const convertedReqScopes = this.trimAndConvertArrayToLowerCase([...reqScopes]);
-            return convertedExtraScopes ? [...convertedReqScopes, ...convertedExtraScopes] : convertedReqScopes;
+            return convertedScopesToAppend ? [...convertedReqScopes, ...convertedScopesToAppend] : convertedReqScopes;
         }
         return null;
     }
 
     // #endregion
 
+    /**
+     * Replaces clientId with openid and profile scopes in the scopes set
+     * for login calls. OIDC required scopes: https://openid.net/specs/openid-connect-basic-1_0.html#Scopes
+     * @param scopes
+     */
+    static generateLoginScopes(scopes: Array<string>, clientId: string): Array<string> {
+        const loginScopes = [...scopes];
+        const clientIdIndex: number = loginScopes.indexOf(clientId);
+        
+        if (clientIdIndex >= 0) {
+            // Splice removes clientId from scope set, which will be replaced by openid and profile
+            loginScopes.splice(clientIdIndex, 1);
+        }
+        if (loginScopes.indexOf(Constants.openidScope) === -1) {
+            loginScopes.push(Constants.openidScope);
+        }
+        if (loginScopes.indexOf(Constants.profileScope) === -1) {
+            loginScopes.push(Constants.profileScope);
+        }
+        
+        return loginScopes;
+    }
+
+    /**
+     * 
+     */
+    static isLoginScopes(scopes: Array<string>, clientId: string): boolean {
```

Comment:
This method was used to avoid adding login scopes to non-login scenarios, I removed the only reference to it in last-minute changes. I think, per my previous answer, that we should remove this method, but I'll think it through a bit more and update the PR.

Reply:
With the latest changes, this method is called within `UserAgentApplication.validateRequest` to determine whether login scopes need to be generated.